python/ray/data/iterator.py

```diff
@@ -295,12 +297,11 @@ def iter_torch_batches(
             collate_fn: A function to apply to each data batch before returning it.
                 Potential use cases include collating along a dimension other than the
                 first, padding sequences of various lengths, or generally handling
-                batches of different length tensors. If not provided, the default
-                collate function is used which simply converts the batch of numpy
-                arrays to a batch of PyTorch tensors. This API is still experimental
+                batches of different length tensors. This API is still experimental
                 and is subject to change.
             finalize_fn: A function to apply to each data batch after it has been
-                collated. This is executed on a GPU-based threadpool if available.
+                collated. If not provided, the default function is used, which
```

Comment:
Please document the concurrency/prefetch behavior of finalize_fn, here and elsewhere.